### PR TITLE
feat: cli arg disable cors torii proxy server

### PR DESCRIPTION
**Pull request by Larkooo**
#1903

### CODE REVIEW

File: bin/torii/src/main.rs [hunk 1]
```diff
@@ -109,6 +109,10 @@ struct Args {
     /// Chunk size of the events page when indexing using events
     #[arg(long, default_value = "1000")]
     events_chunk_size: u64,
+
+    /// Should cors be disabled
+    #[arg(long)]
+    no_cors: bool,
```
Comment (tarrencev) [inline bin/torii/src/main.rs:115]: I think we should rather make `allowed_origins: Vec<String>` `allowed_origins: Option<Vec<String>>` and CORS should be disabled by default

File: crates/torii/server/src/proxy.rs [hunk 2]
```diff
@@ -103,15 +105,19 @@ impl Proxy {
...
-                    origins.iter().map(|o| o.parse().expect("valid origin")).collect::<Vec<_>>(),
-                ),
-            };
+            let cors =
+                allowed_origins.clone().map(|allowed_origins| match allowed_origins.as_slice() {
+                    [origin] if origin == "*" => cors.allow_origin(AllowOrigin::mirror_request()),
+                    origins => cors.allow_origin(
+                        origins
+                            .iter()
+                            .map(|o| o.parse().expect("valid origin"))
```
Comment (glihm) [inline crates/torii/server/src/proxy.rs:114]: This parse will almost never fail, even with `--allowed-origins "fheihf egis g,9435"`. We may want to ensure they are valid URL if not equal to `*`. May be tackled later though depending on the priority.

File: crates/torii/server/src/proxy.rs [hunk 3]
```diff
@@ -103,15 +105,19 @@ impl Proxy {
...
+                            .collect::<Vec<_>>(),
+                    ),
+                });
 
             let graphql_addr_clone = graphql_addr.clone();
-            let service = ServiceBuilder::new().layer(cors).service_fn(move |req| {
+            let service = ServiceBuilder::new().option_layer(cors).service_fn(move |req| {
```
Comment (glihm) [inline crates/torii/server/src/proxy.rs:120]: `cors` is never `None`. We may want to add a `if` to ensure `None` is passed to this function if `allowed_origins` is None.

Comment (Larkooo) [reply crates/torii/server/src/proxy.rs:120]: wdym? the second definition of `let cors` maps the allowed_origins which is an Option<Vec<String>> so it should be None if there are no provided origins

File: crates/torii/server/src/proxy.rs (hunk 2 above)
Comment (Larkooo) [reply crates/torii/server/src/proxy.rs]: Oh yeah nice catch. We'll try to parse it as an URI first

File: crates/torii/server/src/proxy.rs (hunk 3 above)
Comment (glihm) [reply crates/torii/server/src/proxy.rs:124]: Oh, my bad by default it's `*`. Then how can I pass `None`?
I guess by default it's better to have `None` now that we can disable it?

**Review by glihm — APPROVED** [review]

**Pull request merged**